akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -118,25 +133,32 @@ private[akka] abstract class InternalProjectionState[Offset, Envelope](
         case single: SingleHandlerStrategy[Envelope] =>
           val handler = single.handler()
           val handlerRecovery =
-            HandlerRecoveryImpl[Offset, Envelope](projectionId, recoveryStrategy, logger, statusObserver)
+            HandlerRecoveryImpl[Offset, Envelope](projectionId, recoveryStrategy, logger, statusObserver, telemetry)
 
           Flow[ProjectionContextImpl[Offset, Envelope]].mapAsync(parallelism = 1) { context =>
+            val measured: () => Future[Done] = { () =>
+              handler.process(context.envelope).map { done =>
+                // `telemetry.afterProcess` is invoked immediately after `handler.process`
+                // even when retrying. Note that an error in handler.process must not
```

Comment:
is "even when retrying" correct? afterProcess is only after successful processing